src/renderers/dom/shared/DOMProperty.js

```diff
@@ -13,10 +13,26 @@
 
 var invariant = require('fbjs/lib/invariant');
 
+// These attributes should be all lowercase to allow for
+// case insensitive checks
+var RESERVED_PROPS = {
+  children: true,
+  dangerouslysetinnerhtml: true,
+  autofocus: true,
+  defaultvalue: true,
+  defaultchecked: true,
+  innerhtml: true,
```

Comment:
I'm working on making the warning for `onFocusIn` and `onFocusOut` case insensitive. I can do that for innerHTML too.

These show up in `assertValidProps`:
https://github.com/facebook/react/blob/master/src/renderers/dom/shared/utils/assertValidProps.js

The easiest thing to do is move them to `ReactDOMUnknownPropertyHook`.